### PR TITLE
Teach exporter to understand, validate, and respect chunk_size= parameter

### PR DESCRIPTION
**Pull request by ggainey**
Since we can now have multiple-output-files, replaced filename/sha256
columns with output_file_info, a JSONField which is a dictionary of
filename: hash pairs.
closes #6736

### CODE REVIEW

File: pulpcore/app/tasks/export.py [hunk 1]
```diff
@@ -141,62 +144,99 @@ def pulp_export(the_export):
...
-        export_versions(the_export, plugin_version_info)
-        # Export the top-level entities (artifacts and repositories)
-        # Note: we've already handled "what about incrementals" when building the 'artifacts' list
-        export_artifacts(the_export, artifacts)
-        # Export the repository-version data, per-version
-        for version in versions_to_export:
-            export_content(the_export, version)
-            ExportedResource.objects.create(export=the_export, content_object=version)
 
+def compute_hash(filename):
```
Comment (daviddavis) [inline pulpcore/app/tasks/export.py:196]: I'm wondering if this should be `_compute_hash(filename)`?

File: pulpcore/app/tasks/export.py [hunk 2]
```diff
@@ -141,62 +144,99 @@ def pulp_export(the_export):
...
             sha256_hash.update(byte_block)
-        the_export.sha256 = sha256_hash.hexdigest()
-    the_export.filename = tarfile_fp
-    the_export.save()
-    pulp_exporter.last_export = the_export
-    pulp_exporter.save()
+        return sha256_hash.hexdigest()
+
+
+def do_export(pulp_exporter, tar, the_export):
```
Comment (daviddavis) [inline pulpcore/app/tasks/export.py:205]: And `_do_export(...)`?

**Comment by daviddavis** [comment]
Looks good. I think this probably requires a `.removal` file in CHANGES since we are removing/moving fields.

**Comment by ggainey** [comment]
> Looks good. I think this probably requires a `.removal` file in CHANGES since we are removing/moving fields.

Good catch, added!

**Pull request merged**